### ISSUE TITLE
Invariant Violation: Tried to register two views with the same name RNDateTimePicker

### ISSUE THREAD

**Issue by dllz** [GeekyAnts/NativeBase#3649]
## I have gone through these following points

- [x] Check latest documentation: https://docs.nativebase.io/
- [x] Check for existing open/closed issues for a possible duplicate before creating a new issue: https://github.com/GeekyAnts/NativeBase/issues
- [x] Use the latest NativeBase release: thttps://github.com/GeekyAnts/NativeBase/releases
- [] Check examples from NativeBase KitchenSink https://github.com/GeekyAnts/NativeBase-KitchenSink
- [] For discussion purpose make use of NativeBase Slack: http://slack.nativebase.io/
- [] For queries related to theme, check [Theme Variables](https://docs.nativebase.io/docs/ThemeVariables.html) from Docs and live NativeBase Theme Editor http://nativebase.io/customizer/

## Issue Description
When running native base and @react-native-community/datetimepicker v3.2.0 there is a conflict on iOS where the date picker of native-base and datetime picker of the other dependency both try to register themselves as views. As the native-base date-picker does not seem to support time (at least there is no option for it in the documentation) it cannot be used in all circumstances and therefore should play nice with other dependencies. This conflict does not exist on Android. 
The error printed in console is: 
` ERROR  Invariant Violation: Tried to register two views with the same name RNDateTimePicker
 ERROR  Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect. 
      This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native.
 ERROR  Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect. 
      This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native.`
      
 When I remove native base from the project the error goes away and since both dependencies have date pickers I assume that is the cause. 

#### node, npm, react-native, react and native-base version, expo version if used, xcode version
React: 17.0.1
Node: v14.15.4
React-native: 0.64.0
Native-Base: 2.15.2
#### Expected behaviour
The application will run fine with both dependencies.
#### Actual behaviour
Application does not start due to the error
#### Steps to reproduce

```import React from "react";
import {View} from "react-native";
import {Button as NButton, Icon} from "native-base";
import DateTimePicker from "@react-native-community/datetimepicker";
import moment from "moment";

export default class DateTimeSetter extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            date: '',
            time: '',
            showDTPicker: false,
            DTMode: 'date'
        }
        if(this.props.callback) {
            this.props.callback(this.calculateEpoch())
        }
    }

    calculateEpoch() {
        let date;
        let time;
        if(this.state.date == ''){
            date = moment().format("YYYY-MM-DD");
        } else {
            date = this.state.date;
        }
        if(this.state.time == ''){
            time = moment().format("HH:mm:ss");
        } else {
            time = this.state.time;
        }
        console.log(date, time)
        let dateTimestamp = moment(date + ' ' + time, "YYYY-MM-DD HH:mm:ss");
        return dateTimestamp.valueOf();
    }

    render() {
        return (
            <View style={{display: 'flex', flexDirection: 'row'}}>
                <NButton  onPress={() => {
                    this.setState({
                        showDTPicker: true,
                        DTMode: 'date'
                    })
                }} transparent>
                    <Icon name='calendar' style={{color: 'black', fontSize: 30}} type='FontAwesome5'/>
                </NButton>
                <NButton  onPress={() => {
                    this.setState({
                        showDTPicker: true,
                        DTMode: 'time'
                    })
                }} transparent>
                    <Icon name='clock' style={{color: 'black', fontSize: 30}} type='FontAwesome5'/>
                </NButton>
                {this.state.showDTPicker && <DateTimePicker
                    value={new Date()}
                    is24Hour={true}
                    mode={this.state.DTMode}
                    display="default"
                    onChange={(event, dateTime) => {
                        if (dateTime){
                            switch (this.state.DTMode) {
                                case "date":
                                    let date = dateTime.getFullYear() + '-' + (Number(dateTime.getMonth()) + 1) + '-' + dateTime.getDate();
                                    this.setState({
                                        date: date,
                                        showDTPicker: false
                                    })
                                    if(this.props.callback) {
                                        this.props.callback(this.calculateEpoch());
                                    }
                                    break;
                                case "time":
                                    let time = dateTime.toTimeString().split(' ')[0];
                                    this.setState({
                                        time: time,
                                        showDTPicker: false
                                    })
                                    if(this.props.callback) {
                                        this.props.callback(this.calculateEpoch());
                                    }
                                    break;
                            }
                        }
                        console.log("date + time", this.state.date, this.state.time)
                    }}
                />}
            </View>
        )
    }

}


#### Is the bug present in both iOS and Android or in any one of them?
iOS only
#### Any other additional info which would help us debug the issue quicker.

###Important

If you want your issue to be looked at quicker, attach a snack reproducible with your issue. Makes it easier for us!

<!-- In case of not following the above instructions, the issue will be closed abruptly -->

**Comment by aroucha-miguel** [comment]
Had a similar problem causing this error today. I was using `"@react-native-community/datetimepicker": "3.4.6",` on my app package.json. @react-native-community/datetimepicker released a new version `3.4.7` and since native-base is using `^`, it was causing a conflict of versions (as it was using 3.4.7 for native-base and 3.4.6 for my project.

Check wich version you're setting in your package.json. Adding a `^` should be enough to fix this error.

**Comment by nikhilgoswami** [comment]
How to resolve this?
